ui/apps/platform/src/hooks/useURLSort.ts

```diff
@@ -45,65 +34,161 @@ function tableSortOption(
-            ? sortOption.direction
-            : defaultSortOption.direction;
-    const activeAggregateBy =
-        isParsedQs(sortOption) && sortOption?.aggregateBy
-            ? (sortOption?.aggregateBy as SortAggregate)
-            : undefined;
-
-    const internalSortResultOption = useRef<ApiSortOption>(
-        tableSortOption(activeSortField, activeSortDirection, activeAggregateBy)
-    );
-
-    // we'll use this to map the sort fields to an index PatternFly can use internally
-    const [fieldToIndexMap, setFieldToIndexMap] = useState<Record<string, number>>({});
-
-    // we'll construct a map of sort fields to indices that will make it easier to work with
-    // PatternFly
-    useDeepCompareEffect(() => {
-        const newFieldToIndexMap = sortFields.reduce(
-            (acc, curr, index) => {
-                acc[curr] = index;
-                return acc;
-            },
-            {} as Record<string, number>
+    const activeSortOption = getActiveSortOption(sortOption, defaultSortOption);
+
+    const internalSortResultOption = useRef<ApiSortOption>(getValidSortOption(activeSortOption));
+
+    function getSortParams(
+        columnName: string,
+        fieldOptions?: SortAggregate | { field: string; aggregateBy?: SortAggregate }[]
+    ): ThProps['sort'] {
+        // Convert the caller provided sort fields of type (string | string[]) to an
+        // array of string[].
+        const declaredSortFields: string[][] = sortFields.map((field) =>
+            Array.isArray(field) ? field : [field]
         );
-        setFieldToIndexMap(newFieldToIndexMap);
-    }, [sortFields]);
 
-    function getSortParams(field: string, aggregateBy?: SortAggregate): ThProps['sort'] {
-        const index = fieldToIndexMap[field];
-        const activeSortIndex = activeSortField ? fieldToIndexMap[activeSortField] : undefined;
+        // Find the index of the field passed to `getSortParams` in the sortFields provided
+        // to the hook.
+        const targetSortFields: string[] = Array.isArray(fieldOptions)
+            ? fieldOptions.map(({ field }) => field)
+            : [columnName];
+
+        const index = findIndex(
+            declaredSortFields,
+            (field) => intersection(field, targetSortFields).length > 0
+        );
+
+        // Find the index of the active sort field in the declared sort fields.
+        const activeSortOptionFields: string[] = Array.isArray(activeSortOption)
+            ? activeSortOption.map(({ field }) => field)
+            : [activeSortOption.field];
+
+        const activeSortIndex = findIndex(
+            declaredSortFields,
+            (field) => intersection(field, activeSortOptionFields).length > 0
+        );
+
+        // We can't support multiple sort fields with different directions at this time, so
+        // we'll just use the first sort field's direction.
+        const direction = Array.isArray(activeSortOption)
+            ? activeSortOption[0].direction
```

Comment:
Because `getActiveSortOption` has `filter` method, can the array be empty here?

Reply:
Which reminds me to ask, is empty array `[]` the counterpart for multiple sort to `defaultSortOption` for single sort?

Reply:
Good catch here - I've updated the hook's API to specify that a valid multi sort is a `NonEmptyArray<FieldOption>` as opposed to a `FieldOption[]`. In the case where an empty array somehow slips in (e.g. via a JS file), the default option will be used instead.